### DISCUSSION TITLE
Is it possible to set `rangeStrategy` per `matchUpdateTypes`

### DISCUSSION THREAD

**Discussion by mtmeyer** [renovatebot/renovate#16407] [Request Help] [answered]
I was hoping to set the `rangeStragegy` for major updates to be different to minor and patch updates.  I've tried this a few different ways in the config (using [minor](https://docs.renovatebot.com/configuration-options/#minor) & [patch](https://docs.renovatebot.com/configuration-options/#patch) config and also the [packageRules](https://docs.renovatebot.com/configuration-options/#packagerules)).  I noticed in [this issue](https://github.com/renovatebot/renovate/discussions/14369#discussioncomment-2235354) that `matchUpdateTypes` happens after `rangeStrategy` in the `packageRules` and was wondering if this is also the case for the `minor` and `patch` configuration options.

TLDR: Is it possible to split `rangeStrategy` based on the update type (major, minor, patch)

**Comment by rarkins** [comment, marked as answer]
No, because range strategy is used to decide major/minor. What are you trying to achieve?

**Reply by mtmeyer** [reply]
We're looking for minor and patch updates to only change the lockfile whereas major should result in a package.json change.  
Was thinking for minor & patch it would be `in-range-only` and for major `replace`.
Reading the documentation again for `update-lockfile` am i right in staying that this will actually achieve what i'm looking to do?

**Reply by rarkins** [reply]
How would that work? If a minor upgrade *needs* to update the package.json then there's no away around that

**Reply by mtmeyer** [reply]
Good point.  So am I correct in saying that with `update-lockfile`, if its within the acceptable range as determined by `package.json` (i.e. "^1.4.0" to 1.5.0), then it will only update the lockfile.  If it's outside the acceptable range (i.e. "1.4.0" to 1.5.0) then it will update `package.json`.

**Reply by rarkins** [reply]
Yes, that's right. You could think of it like this:
- If it can update without changing lock file then it will, and leave package.json untouched
- If it can't update without also changing package.json, then it will do that too (essentially same as rangeStrategy=replace)

**Reply by mtmeyer** [reply]
Awesome, cheers for the help, really appreciate it 🙏
